scripts/update_nmdc_submission_schema.py

```diff
@@ -0,0 +1,43 @@
+import json
+from contextlib import contextmanager
+import requests
+
+@contextmanager
```

Comment:
Right, the crux of the issue here is exactly that: it's a really big file. Currently in dev when it's sent over the wire it's 3.3 MB. To somewhat mitigate the cost of sending it we currently delay fetching it until it's actually needed by using a dynamic import: https://github.com/microbiomedata/nmdc-server/blob/7ae56de2c5b3394a535985ed9dffd576712359fa/web/src/views/SubmissionPortal/harmonizerApi.ts#L141
That way someone just browsing the data portal doesn't incur that load time. 

This script attempts to address two things. First, the schema is currently checked in alongside the code with no obvious mechanism for updating it. But also, in addition to fetching a specific revision of the submission schema, this script jettisons some bits of the schema that aren't needed by DataHarmonizer. It slims it the resulting JSON down to 1.7 MB (it might be a little less than that in a production build) -- still big, but not an insignificant drop. 

To your point, we definitely could apply the same strategy that we use to incorporate the main schema to the submission schema as well (i.e. using a GitHub URL to https://github.com/microbiomedata/sheets_and_friends as a dependency in `package.json`). That would be a win for uniform dependency management, but we'd have to forgo attempting to minimize the schema size.

Taking a step back, I think the `sheets_and_friends` build process could be improved in a number of ways including reducing the JSON file size, getting proper versioning sorted out, publishing artifacts to NPM, etc. I guess my preference would be to leave these changes in place while we work on some of those upstream issues. But I'm happy to continue discussing it.

Reply:
I'm fine with these changes being in place for now. There's another possible solution that is presented in [this issue](https://github.com/microbiomedata/nmdc-server/issues/794). I think any changes to address the bigger problem are likely out of scope of this PR. Thanks for your explanation here, it was helpful.